### ISSUE TITLE
graphql-config related performance improvement for processor

### ISSUE THREAD

**Issue by lacco** [B2o5T/graphql-eslint#1195]
**Is your feature request related to a problem? Please describe.**

In my project, we have a lot of GraphQL operations embedded in `*.ts` files via `gql` tag.
That is why we have a configuration like this:

```
{
  files: '*.ts',
  processor: '@graphql-eslint/graphql',
}
```

Adding the above configuration slows down the execution of `eslint .` from 5s to 15s.
We are not using `graphql-config`, therefore no such file exists.
When adding a graphl config file, the execution goes down to 7s, which is much better.

The processor tries to cache the graphql config: https://github.com/B2o5T/graphql-eslint/blob/master/packages/plugin/src/processor.ts#L19
If the config is missing, `onDiskConfig` will always stay `undefined`.
For each file, the processor tries to load the configuration again.

**Describe the solution you'd like**

I can imagine a couple of solutions:

1. Add configuration option to turn off use of usage of `graphql-config`
2. Introduce an explicit boolean `onDiskConfigLoaded`
```
let onDiskConfig: GraphQLConfig;
let onDiskConfigLoaded = false;

// ...
  preprocess(code, filePath) {
      if(!onDiskConfigLoaded) {
        onDiskConfig = loadOnDiskGraphQLConfig(filePath);
        onDiskConfigLoaded = true
      }
```
3. change documentation that a graphql config file is recommended for performance reasons

**Describe alternatives you've considered**

Adding a graphql config file fixes the performance issue.

**Comment by B2o5T** [comment]
Amazing catch! Thanks ❤️ will fix this week

**Comment by B2o5T** [comment]
Could you test `3.11.3-alpha-20221008220048-6ed3933` and say if your issue was fixed?

**Comment by B2o5T** [comment]
should be fixed in `@graphql-eslint/eslint-plugin@3.12.0`

**Comment by lacco** [comment]
@B2o5T Amazing work! 
Confirmed that `3.12.0` is much faster when using the processor without a `graphl-config` file ❤️
